### ISSUE TITLE
Segmentation fault on when passing context to addFunction

### ISSUE THREAD

**Issue by genuss** [wcgallego/pecl-gearman#49]
It's possible to pass context object as the third parameter to GearmanWorker::addFunction(). If this context is an object or an array then segmentation fault will occur after doing first job.
Here is a minimal test example:
TestWorker.php:
````php
<?php

class TestWorker extends GearmanWorker
{
    public function __construct()
    {
        parent::__construct();
        $this->addServer();
        $this->addFunction('test', [$this, 'test'], ['firstArg' => 'firstValue']);
    }

    public function test($job, $context)
    {
        echo "Starting job {$job->workload()}". PHP_EOL;
        $firstArg = $context['firstArg'];
        echo "FirstArg is $firstArg" . PHP_EOL;
    }
}

$testWorker = new TestWorker();
while ($testWorker->work()) {};

````
TestClient.php:
````php
<?php

$client = new GearmanClient();
$client->addServer();
$client->doBackground('test', 'job1');
$client->doBackground('test', 'job2');
````
Assuming gearman server is on 127.0.0.1:4730 run
````
php TestClient.php
php TestWorker.php
````
The output will be:
````
Starting job job1
FirstArg is firstValue
Starting job job2
Segmentation fault (core dumped)
````
PHP version: 7.0.15
OS: Debian 8 jessie
pecl-gearman version: 2.0.3
libgearman version: 1.0.14
gearman server version: 1.0.14

**Comment by wcgallego** [comment]
Thanks @genuss, this was one of those darker corners of the code that didn't get as much testing. Appreciate the thorough comment. I'll take a look

**Comment by wcgallego** [comment]
sorry for 3+ week delay in this @genuss, life gets in the way. Just tested it out on my instance - able to replicate the seg fault before and confirmed the fix is good. Much appreciated!

Whoops, forgot to mention - the fix from @muzki, pull request #52

**Comment by genuss** [comment]
@wcgallego and @muzki thanks you guys! That's very nice to see the issue fixed.
Will try on my installation asap.
